### ISSUE TITLE
Weight changing of prompt part without () in a prompt with AND

### ISSUE THREAD

**Issue by Nialpo** [yownas/shift-attention#1]
With your script, if I make a prompt like
"photo of (cat:1~ 0) or (dog:0~ 1) AND green background:0~1"   (ignore spaces)
The script don't change last weight. 
Supposed to be:
"photo of (cat:1) or (dog:0) AND green background:0"
"photo of (cat:0.9) or (dog:0.1) AND green background:0.1"
...etc

And "photo of (cat:1~ 0) or (dog:0~ 1) AND (green background:0~1)" gives greatly different output... But it is a thing of SD itself

**Comment by yownas** [comment]
Yes. That is not an error. The script only change things in the pattern ":<number>~<number)".

I haven't used AND before, but would adding a number at the end work? In that case I can see if I can make the script work without ")".

**Comment by Nialpo** [comment]
>I haven't used AND before, but would adding a number at the end work?

Yes, and before AND too. Examples here: (3.) https://www.youtube.com/watch?v=Ch3jsq6O2cI
Thanks
